src/mainwindow.cpp

```diff
@@ -36,22 +36,29 @@ MainWindow::MainWindow(QWidget *parent) :
     ui->baudRate->setCurrentIndex(3);
     loadPortsAndSet();
 
-    connect(this, &MainWindow::startMonitor, m_monitor, &Monitor::readAndParse);
-    m_monitor->moveToThread(&m_monitorThread);
-    m_monitorThread.start();
-
-    connect(ui->monitorButton, &QToolButton::toggled, this, &MainWindow::handleMonitorToggle);
+    connect(ui->monitorButton, &QToolButton::toggled, this, &MainWindow::handleMonitorToggled);
     connect(ui->clearButton, &QToolButton::released, ui->plainTextEdit, &QPlainTextEdit::clear);
     connect(ui->sendButton, &QToolButton::released, this, &MainWindow::handleSend);
     connect(ui->portReload, &QToolButton::released, this, &MainWindow::handleReloadPorts);
     connect(ui->port, static_cast<void (QComboBox::*)(int index)>(&QComboBox::currentIndexChanged), this, &MainWindow::handlePortChanged);
-    connect(this, &MainWindow::open, m_monitor, &Monitor::open);
     connect(ui->baudRate, static_cast<void (QComboBox::*)(int index)>(&QComboBox::currentIndexChanged), this, &MainWindow::handleBaudRateChanged);
```

Comment:
I'm curious to know why these need to be casted

Reply:
Well, so QComboBox::currentIndexChanged is an overloaded signal with two variants, one which emits the current index (int), and another which emits the current text (QString). I just saw in the documentation that there's a cleaner more "Qt" way to do it, which uses this instead: `QOverload<int>::of(&QComboBox::currentIndexChanged)`, so I'll change this to that so it's clearer